numpyro/distributions/distribution.py

```diff
@@ -313,6 +313,9 @@ def __init__(self, base_dist, batch_shape=()):
         if isinstance(base_dist, ExpandedDistribution):
             base_dist = base_dist.base_dist
         self.base_dist = base_dist
+        if isinstance(base_dist, ExpandedDistribution):
+            batch_shape = self._broadcast_shape(base_dist.batch_shape, batch_shape)
+            base_dist = base_dist.base_dist
```

Comment:
This optimization is addressed above. Do you think we need extra logic here?

Reply:
I missed that during merge, I think we should still do an explicit broadcast so that. `dist.expand(5, 4).expand(6, 4)` raises an error. Will fix.

Reply:
Hmm, `.expand(5, 4).expand(6, 4)` seems to be invalid, hence an error is expected. Did you mean something like: `.expand(5, 1).expand(1, 4)`?

Reply:
You are right!!